evergreen/compile-centos7.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# This script runs in a container built from Dockerfile-centos7.
+# It assumes the workspace directory is mounted on /etc/envoy-serverless and the build directory is optionally mount on /build
+
+set -euo pipefail
+set -x
+
+# Use MongoDB toolchain installed in the docker image.
+export LD_LIBRARY_PATH="/opt/mongodbtoolchain/v4/lib"
+export CC=/opt/mongodbtoolchain/v4/bin/clang
+export CXX=/opt/mongodbtoolchain/v4/bin/clang++
+
+# Test targets listed in repo's .bazelci config
+export bazel_tests="//test/common/common/... //test/integration/... //test/exe/..."
+# Include serverless specific patches.
+export bazel_tests="${bazel_tests} //test/extensions/filters/listener/proxy_protocol:proxy_protocol_test //test/extensions/common/proxy_protocol:proxy_protocol_header_test //test/extensions/transport_sockets/proxy_protocol:proxy_protocol_integration_test"
+# This test started failing in 1.17, but only when run by evergreen. not reproducible
+# in any environment other than being executed by jasper/evergreen curator.
+export bazel_tests="${bazel_tests} -//test/common/common:logger_speed_test_benchmark_test"
+export bazel_flags="--config=linux --config=clang"
+export bazel_flags="${bazel_flags} -c opt --action_env=PATH --action_env=CC=${CC} --action_env=CXX=${CXX} --action_env=LD_LIBRARY_PATH --verbose_failures -k"
```

Comment:
According to Bazel's documentation, `--config=clang` at line 21 expands the option group called `clang` in place, which specifies `--action_env=CC=clang`, so a following `--action_env=CC` at line 22 should be able to override it using the value in the environment variable `CC`. However it doesn't seem to work for some external dependencies, where `CC` is correctly set to the MongoDB toolchain, but the system `clang` was called for compile. Eventually, I had to override with `--action_env=CC=${CC}`.

Reply:
Ugh.  Understood.